src/main/java/org/zalando/nakadi/validation/schema/diff/CombinedSchemaDiff.java

```diff
@@ -17,8 +17,7 @@
     static void recursiveCheck(final CombinedSchema combinedSchemaOriginal, final CombinedSchema combinedSchemaUpdate,
                                        final Stack<String> jsonPath,
                                        final List<SchemaChange> changes) {
-        if(!(combinedSchemaOriginal.getSubschemas().containsAll(combinedSchemaUpdate.getSubschemas())
-                && combinedSchemaUpdate.getSubschemas().containsAll(combinedSchemaOriginal.getSubschemas()))) {
+        if(combinedSchemaOriginal.getSubschemas().size() != combinedSchemaUpdate.getSubschemas().size()) {
```

Comment:
Is there any chance that both list contain the same number of elements, but that they are different?